libraries/eosiolib/core/eosio/datastream.hpp

```diff
@@ -765,6 +765,46 @@ datastream<Stream>& operator >> ( datastream<Stream>& ds, std::vector<T>& v ) {
    return ds;
 }
 
+/**
+ *  Serialize a basic_string
+ *
+ *  @param ds - The stream to write
+ *  @param s - The value to serialize
+ *  @tparam Stream - Type of datastream buffer
+ *  @tparam T - Type of the object contained in the basic_string
+ *  @return datastream<Stream>& - Reference to the datastream
+ */
+template<typename Stream, typename T>
+datastream<Stream>& operator << ( datastream<Stream>& ds, const std::basic_string<T>& s ) {
+   ds << unsigned_int( s.size() );
+   for( const auto& i : s ) {
```

Comment:
Wouldn't `datastream.write` and `datastream.read` below be better?

Reply:
I checked in terms of performance and (if I'm reading assembler code correctly) there should be no difference for operator<< with above `for` loop vs `ds.write` - both end with a single call to memcpy (when compiled with `-O2` flag)

From readers perspective ds.write/read is a bit more concise so I have changed the code.

basic_string operator<< with for loop:
![basic_string_for](https://user-images.githubusercontent.com/2914819/227222386-90ac3ea4-b4a6-48dd-8e1c-97257bcd2718.png)
basic_string operator<< with ds.write:
![basic_string_write](https://user-images.githubusercontent.com/2914819/227222500-dfe4a77b-898c-4de7-ae78-f5642cc9d884.png)

Reply:
operator>> and ds.read needs `sizeof(T)`, but `sizeof` is a compile time thing (there is no runtime cost) so I wouldn't worry about it.
I think it is only matter of what is easier to read by developer.